zcash_primitives/src/zip32.rs

```diff
@@ -134,6 +134,27 @@ impl Default for DiversifierIndex {
     }
 }
 
+impl From<u32> for DiversifierIndex {
+    fn from(i: u32) -> Self {
+        u64::from(i).into()
+    }
+}
+
+impl From<u64> for DiversifierIndex {
+    fn from(i: u64) -> Self {
```

Comment:
shouldn't these be `pub fn` ?

Reply:
No; trait implementation methods are public by default.